src/platform/ESP32/NetworkCommissioningDriver.cpp

```diff
@@ -177,23 +177,48 @@ void ESPWiFiDriver::OnConnectWiFiNetwork()
 {
     if (mpConnectCallback)
     {
+        DeviceLayer::SystemLayer().CancelTimer(ConnectWifiNetworkTimeout, NULL);
         mpConnectCallback->OnResult(Status::kSuccess, CharSpan(), 0);
         mpConnectCallback = nullptr;
     }
 }
 
+void ESPWiFiDriver::OnConnectWiFiNetworkFailed()
+{
+    if (mpConnectCallback)
+    {
+        mpConnectCallback->OnResult(Status::kNetworkNotFound, CharSpan(), 0);
+        mpConnectCallback = nullptr;
+    }
+}
+
+void ESPWiFiDriver::ConnectWifiNetworkTimeout(chip::System::Layer * aLayer, void * aAppState)
+{
+    CHIP_ERROR error = chip::DeviceLayer::Internal::ESP32Utils::ClearWiFiStationProvision();
+    if (error != CHIP_NO_ERROR)
+    {
+        ChipLogError(DeviceLayer, "ClearWiFiStationProvision failed: %s", chip::ErrorStr(error));
+    }
+    NetworkCommissioning::ESPWiFiDriver::GetInstance().OnConnectWiFiNetworkFailed();
+}
+
 void ESPWiFiDriver::ConnectNetwork(ByteSpan networkId, ConnectCallback * callback)
 {
-    CHIP_ERROR err          = CHIP_NO_ERROR;
-    Status networkingStatus = Status::kSuccess;
+    CHIP_ERROR err              = CHIP_NO_ERROR;
+    Status networkingStatus     = Status::kSuccess;
+    const uint32_t secToMiliSec = 1000;
 
     VerifyOrExit(NetworkMatch(mStagingNetwork, networkId), networkingStatus = Status::kNetworkIDNotFound);
     VerifyOrExit(mpConnectCallback == nullptr, networkingStatus = Status::kUnknownError);
     ChipLogProgress(NetworkProvisioning, "ESP NetworkCommissioningDelegate: SSID: %.*s", static_cast<int>(networkId.size()),
                     networkId.data());
 
-    err               = ConnectWiFiNetwork(reinterpret_cast<const char *>(mStagingNetwork.ssid), mStagingNetwork.ssidLen,
+    err = ConnectWiFiNetwork(reinterpret_cast<const char *>(mStagingNetwork.ssid), mStagingNetwork.ssidLen,
                              reinterpret_cast<const char *>(mStagingNetwork.credentials), mStagingNetwork.credentialsLen);
+
+    err = DeviceLayer::SystemLayer().StartTimer(
+        static_cast<System::Clock::Timeout>((kWiFiConnectNetworkTimeoutSeconds - 1) * secToMiliSec), ConnectWifiNetworkTimeout,
```

Comment:
The device can't guarantee delivery of the failure notification within ConnectMaxTimeSeconds, even if we subtract 1 here.  So why would we try?  Shouldn't chip-tool know enough to wait slightly longer than this amount of time?  I think this adjustment is in the wrong code and needs to be in the client (chip-tool) instead.

Also, why is this unrecoverable?  It would be nice if expiration of the fail-safe context would cleanup network config in this platform.

Reply:
Increased chip-tool timeout by 1 sec. Cleanups will be at failSafe cleanups.